modules/core/shared/src/main/scala/lucuma/core/math/ApparentRadialVelocity.scala

```diff
@@ -22,10 +22,10 @@ final case class ApparentRadialVelocity(cz: Quantity[BigDecimal, MetersPerSecond
   /**
     * Converts the apparent radial velocity to a Redshift
     */
-  def toRedshift: Redshift = Redshift((cz / SpeedOfLight).value)
+  def toRedshift: Redshift = Redshift((cz / constant[BigDecimal, SpeedOfLight]).value)
 
   override def toString =
-    s"ApparentRadialVelocity(${cz.to[Double, KilometersPerSecond].show})"
+    s"ApparentRadialVelocity(${cz.toUnit[KilometersPerSecond].toValue[Double].show})"
```

Comment:
@erikerlandson I'm getting a compile error here, that I can't really understand.
```
Found:    (ApparentRadialVelocity.this.cz : 
  coulomb.Quantity[BigDecimal, lucuma.core.math.units.MetersPerSecond]
)
Required: qopaque$_this.Quantity²[BigDecimal, coulomb.units².accepted.Meter / 
  coulomb.units².accepted.Second
]

where:    Quantity  is a type in package object quantity$package which is an alias of [V, U] =>> coulomb.qopaque.Quantity[V, U]
          Quantity² is a type in object qopaque with bounds <: [V, U] =>> Any
          units     is a object in package lucuma.core.math
          units²    is a package in package coulomb
```

Reply:
@erikerlandson I also have a guess about this one. It looks like a repeat of the `export` issue, `coulomb.Quantity` is the exported version of `qopaque$_this.Quantity`.

Reply:
@armanbilge  It looks like it ought to work, but I'm wondering if it is another problem relating somehow to type aliases.  

The other thing that might be suspect is doing all that in `toString` - in my experience it often doesn't work well with implicits, although usually when that is happening it will complain about a bad method signature or missing `using` arguments.

Reply:
One other possibility is that you are not importing all the unit defs you need for scala to resolve the types all the way down to base-units.

Reply:
@erikerlandson I was able to fix it by switching all `coulomb.*` imports to `import coulomb.qopaque.{Quantity, withUnit}` instead. I really think `export` just doesn't work.

Reply:
I still consider it an option to go back to `AnyVal`.  I liked the idea of an opaque type because it seemed like a better conceptual fit, but if it is going to cause these hard-to-predict and hard-to-diagnose failures then maybe it isn't a good trade.

Reply:
> What exactly was broken?

I no longer remember exactly but trying to put it at the top level of the package was failing to compile.  Possibly there is a way to make it work that I failed to see?

Since then I ran across another reason:  when putting the extensions at the top level, the compiler was failing to distinguish between something like `.show` in `Quantity` versus `DeltaQuantity`. Putting the defs and extensions under `qopaque` and `dqopaque` fixed that.

Except it's still breaking...

Reply:
it would be nice to minimize or eliminate boxing, but OTOH it's already boxing in a lot of places, and I am becoming concerned with the current opaque-type representation being a minefield of compilation problems.

Reply:
xref on opaque boxing: https://github.com/lampepfl/dotty/discussions/12011

Reply:
aside: my dream for `coulomb` is that - at the byte code level - the actual class `Quantity` never appears, or even the various type-classes - it all eventually compiles down to things like `x + c y`, where `c` is some conversion constant the compilation process derived and threw everything else away after it was finished with it.

My hope was that `opaque type` would move me closer to that dream, but the state of the jvm and scala compiler aren't there yet.

Reply:
If I want to make real progress on `coulomb` I simply don't have the spare time to be fighting obscure battles with the compiler on fundamental types like `Quantity` and its operations.

Reply:
Possibly I could just stop supporting the `export qopaque.Quantity` and friends, and you just have to `import coulomb.quantity.*` or something similar.  If that works it would be a reasonable compromise that keeps the opaque type.

Reply:
Right, that's effectively what we are doing now anyway.

Reply:
I'm reading this:
https://dotty.epfl.ch/docs/reference/contextual/extension-methods.html#translation-of-calls-to-extension-methods

And my interpretation is that I should be able to define my extension inside companion `object Quantity`,
and those methods should be visible if I `import coulomb.Quantity`, however it doesn't seem to be working.

Wondering if it is yet another situation where `opaque` has bugs

Reply:
> And my interpretation is that I should be able to define my extension inside companion `object Quantity`,
> and those methods should be visible if I `import coulomb.Quantity`, however it doesn't seem to be working.

Huh, it _should_ work like that. Here's an example from one of my projects:
https://github.com/armanbilge/schrodinger/blob/398e21ceadd862a61252f69559b9b074f6e2582d/math/src/main/scala/schrodinger/math/LogDouble.scala#L23

Reply:
I'm getting problems around `def value`, where it seems to be confusing `V` with `Quantity[V, _]`

Reply:
That's the same as the `log` method in mine:
https://github.com/armanbilge/schrodinger/blob/398e21ceadd862a61252f69559b9b074f6e2582d/math/src/main/scala/schrodinger/math/LogDouble.scala#L57

Reply:
```
  |value value is not a member of coulomb.quantity$package.Quantity[Double, String]
```

getting errors like this.  I notice you put your LogDouble in a file called LogDouble.scala, and mine is in `quantity.scala` - I am wondering if I am "supposed" to use Quantity.scala

Reply:
@erikerlandson can you put up a branch for me to look at?

Reply:
I feel like I'm randomly thrashing around.  something is wrong but I can't quite center on it

Reply:
> can you put up a branch for me to look at?
Yes, let me start over clean, and put stuff in `Quantity` companion object, and I will push